EDMarketConnector.py

```diff
@@ -330,6 +330,30 @@ def __init__(self, master: tk.Tk):  # noqa: C901, CCR001 # TODO - can possibly f
 
         self.prefsdialog = None
 
+        if platform == 'win32':
+            import threading
+            from infi.systray import SysTrayIcon
+
+            self.only_tray_close = 0  # This is kind of a hack.
+            # When the tray icon is double click to reopen the EDMC window, I want the tray icon to disappear. To do
+            # that I have to call the `shutdown()` method of the systray object. Calling the shutdown method triggers
+            # the method associated with `on_quit` twice, once for WM_DESTROY and again for WM_CLOSE. This is not the
+            # case when the application is exited by clicking on `Quit` in the tray menu. So, to handle this, we are
+            # creating this class variable and setting it to 2 when `Open` is called from the tray menu (either by
+            # clicking open or by double clicking the icon) and decrementing it by 1 when the tray shutdown is triggered
+
+            def open_window(systray) -> None:
+                self.only_tray_close = 2
```

Comment:
At the very least, if we agree to keep this hacky behaviour then the possible values of `self.only_tray_close` need to be defined in an Enum and used as such.  No magic numbers in actual code.

Reply:
Hacky behaviour no longer there